src/meta/Macro.h

```diff
@@ -168,6 +169,30 @@ class Macro {
     return str;
   }
 
+  static inline std::vector<std::string> enumeratePathsWithCustomMacros(const std::string& input, const std::map<std::string, std::vector<std::string>>& macroValues, int macroIndex = 0) {
```

Comment:
can we add a new unit test for this newly added core method?

I suggest adding at least one function in this file https://github.com/varchar-io/nebula/blob/master/src/meta/test/TestMacro.cpp